src/devices/scmplus.c

```diff
@@ -0,0 +1,198 @@
+
+    // endpoint_type = b[3];
+    snprintf(endpoint_type_str, sizeof(endpoint_type_str), "0x%02X", b[3]); // endpoint_type);  // b[3]
+
+    memcpy(&t_32, &b[4], 4);
+    endpoint_id = ntohl(t_32);
+
+    memcpy(&t_32, &b[8], 4);
+    consumption_data = ntohl(t_32);
+    //  consumption_data = ((uint32_t)b[8] << 24) | (b[9] << 16) | (b[10] << 8) | (b[11]);
+
+    memcpy(&t_16, &b[12], 2);
+    physical_tamper = ntohs(t_16);
+    snprintf(physical_tamper_str, sizeof(physical_tamper_str), "0x%04X", physical_tamper);
+
+    snprintf(crc_str, sizeof(crc_str), "0x%04X", crc);
+
+    /*
+    if (decoder->verbose && 0) {
+        (void)fprintf(stderr, "protocol_id = %d %02X\n", protocol_id,protocol_id);
+        bitrow_printf(&b[3], 8, "%s\t%2d\t%02X\t", "endpoint_type   ", endpoint_type, endpoint_type);
+        bitrow_printf(&b[4], 32, "%s\t%2d\t%02X\t", "endpoint_id    ", endpoint_id, endpoint_id);
+        bitrow_printf(&b[8], 32, "%s\t%2d\t%02X\t", "consumption_data", consumption_data, consumption_data);
+        // (void)fprintf(stderr, "consumption_data = %d %08X\n", consumption_data,consumption_data);
+        (void)fprintf(stderr, "physical_tamper = %d %04X\n", physical_tamper,physical_tamper);
+        (void)fprintf(stderr, "pkt_checksum = %d %04X\n", pkt_checksum,pkt_checksum);
+    }
+    */
+
+    // Least significant nibble of endpoint_type is  equivalent to SCM's endpoint type field
+    // id info from https://github.com/bemasher/rtlamr/wiki/Compatible-Meters
+    char *meter_type;
+    switch (b[3] & 0x0f) {
+        case 4:
+        case 5:
+        case 7:
+        case 8:
+            meter_type = "Electric";
+            break;
+        case 2:
+        case 9:
+        case 12:
+            meter_type = "Gas";
+            break;
+        case 11:
+        case 13:
+            meter_type = "Water";
+            break;
+        default:
+            meter_type = "unknown";
+            break;
+    }
+    // (void)fprintf(stderr, "meter_type = %s\n", meter_type);
+
+    /* 
+        Field key names and format set to  match rtlamr field names 
+
+        {Time:2020-06-20T09:58:19.074 Offset:49152 Length:49152
+        SCM+:{ProtocolID:0x1E EndpointType:0xAB EndpointID:  68211547 Consumption:  6883 Tamper:0x4900 PacketCRC:0x39BE}}
+    */
+
+    /* clang-format off */
+    data = data_make(
+            "model",           "",                 DATA_STRING, "SCM+",
+            "ProtocolID",     "Protocol_ID",       DATA_STRING, protocol_id_str,
+            "EndpointType",   "Endpoint_Type",     DATA_STRING, endpoint_type_str,
+            "EndpointID",     "Endpoint_ID",       DATA_INT,    endpoint_id,
+            "Consumption",     "",                 DATA_INT,    consumption_data,
+            "Tamper",          "",                 DATA_STRING, physical_tamper_str,
+            "PacketCRC",       "crc",              DATA_STRING, crc_str,
```

Comment:
This value is not really needed but can be kept for rtlamr compatibility reasons.

Reply:
yes it was included for theoretical  compatibility.

compatibility is also the reason I convert the fields to hex strings manually as opposed to using DATA_FORMAT